Source/diablo.cpp

```diff
@@ -587,20 +585,22 @@ static void LoadOptions()
 	sgOptions.Audio.nBufferSize = getIniInt("Audio", "Buffer Size", DEFAULT_AUDIO_BUFFER_SIZE);
 	sgOptions.Audio.nResamplingQuality = getIniInt("Audio", "Resampling Quality", DEFAULT_AUDIO_RESAMPLING_QUALITY);
 
-#ifndef __vita__
 	sgOptions.Graphics.nWidth = getIniInt("Graphics", "Width", DEFAULT_WIDTH);
 	sgOptions.Graphics.nHeight = getIniInt("Graphics", "Height", DEFAULT_HEIGHT);
-#else
-	sgOptions.Graphics.nWidth = DEFAULT_WIDTH;
-	sgOptions.Graphics.nHeight = DEFAULT_HEIGHT;
-#endif
-	sgOptions.Graphics.bFullscreen = getIniBool("Graphics", "Fullscreen", true);
 #if !defined(USE_SDL1) && !defined(__vita__)
 	sgOptions.Graphics.bUpscale = getIniBool("Graphics", "Upscale", true);
+#elif defined(__vita__)
+	sgOptions.Graphics.bUpscale = true;
 #else
 	sgOptions.Graphics.bUpscale = false;
 #endif
+#ifndef __vita__
+	sgOptions.Graphics.bFullscreen = getIniBool("Graphics", "Fullscreen", true);
 	sgOptions.Graphics.bFitToScreen = getIniBool("Graphics", "Fit to Screen", true);
+#else
+	sgOptions.Graphics.bFitToScreen = true;

```

Comment:
Why not leave this as configurable? Same with bUpscale.

Reply:
That currently leads to stretched image instead of black bars.
For upscale: as far as i understood from wiki, turning upscale off will lead to trying to set window/fullscreen resolution to the internal one, which is impossible on vita right now

Reply:
Hmm yeah I think under SDL2 it would actually give you the same internal scaling as the native one. For SDL1 you should get black bars on all sides.